src/lib/extension-load-queue.ts

```diff
@@ -1,63 +1,149 @@
+ *
+ * @param {symbol} storeKey - The key associated with the script store.
+ * @returns {ScriptStore<T>} The script store associated with the given key.
+ */
+export const getScriptStore = <T>(storeKey: symbol): ScriptStore<T> => {
     if (isBrowser()) {
-        (window as Window)[key] = (window as Window)[key] || [];
-        return (window as Window)[key];
+        // Initialize the store if it does not exist or check that it is an array
+        if (!window[storeKey]) {
+            window[storeKey] = [];
+        } else if (!Array.isArray(window[storeKey])) {
+            throw new Error(`Expected window[${String(storeKey)}] to be an array`);
+        }
+
+        return window[storeKey] as ScriptStore<T>;
     } else {
         throw new Error('This functionality should be employed on the client-side.');
     }
 };
 
-export const getQueueStore = () => {
+/**
+ * Ensures that the queues are initialized as an object in the global window.
+ *
+ * @returns {void}
+ */
+const ensureQueuesSymbolInitialized = (): void => {
     if (isBrowser()) {
-        (window as Window)[QUEUE_SYMBOL] = (window as Window)[QUEUE_SYMBOL] || false;
-        return (window as Window)[QUEUE_SYMBOL];
+        if (typeof window[QUEUES_SYMBOL] !== 'object' || window[QUEUES_SYMBOL] === null) {
+            window[QUEUES_SYMBOL] = {};
+        }
+    } else {
+        throw new Error('This functionality should be employed on the client-side.');
     }
+};
 
-    return null;
+/**
+ * Retrieves the status of whether the queue associated with the queueKey is created.
+ *
+ * @param {symbol} queueKey - The key associated with the queue.
+ * @returns {boolean} Whether the queue is created.
+ */
+export const getQueueStore = (queueKey: symbol): boolean => {
+    ensureQueuesSymbolInitialized();
+    return window[QUEUES_SYMBOL]?.[queueKey] || false;
 };
 
-export const handleQueueCreated = (created: boolean) => {
-    (window as Window)[QUEUE_SYMBOL] = created;
+/**
+ * Returns a function to mark the queue as created for a given queueKey.
+ *
+ * @param {symbol} queueKey - The key associated with the queue.
+ * @returns {function(boolean): void} A function that takes a boolean `created`
+ * and marks the queue as created.
+ */
+export const createHandleQueueCreated = (queueKey: symbol): ((created: boolean) => void) => {
+    ensureQueuesSymbolInitialized();
+    return (created: boolean) => {
+        window[QUEUES_SYMBOL][queueKey] = created;
+    };
 };
 
+/**
+ * Creates and manages a load queue.
+ *
+ * @param {CreateLoadQueueArgs<T>} args - The arguments required to create a load queue.
+ * @returns {void}
+ */
 export const createLoadQueue = <T>({
     store,
     createController,
-    isQueueCreated = getQueueStore(),
-    onQueueCreated = handleQueueCreated,
-}: CreateLoadQueueArgs<T>) => {
+    queueKey = SINGLE_QUEUE_SYMBOL,
+    isQueueCreated = getQueueStore(queueKey),
+    onQueueCreated = createHandleQueueCreated(queueKey),
```

Comment:
I can't say I like the fact we can pass in custom `onQueueCreated`, which would effectively break the whole system — the default handler does a very important job of tracking which queues were initialized. So my guess is we don't want to expose this option to the end user.

Reply:
I did not want to associate the `createLoadQueue` with the `window` runtime, assuming that the queue logic could generally work with any data source (store). I suggested leaving the prop `onQueueCreated`. 

However, I agree that `createHandleQueueCreated` function could not be exported.